app/models/tag.rb

```diff
@@ -54,8 +53,17 @@ def taggings_count_cache_key
   end
 
   def write_taggings_to_redis(value)
+    # Atomically set the value while extracting the old value.
+    old_redis_value = REDIS_GENERAL.getset("tag_update_#{id}_value", value)
+
+    # If the value hasn't changed from the saved version or the REDIS version,
+    # there's no need to write an update to the database, so let's just bail
+    # out.
+    return value if value == old_redis_value && value == taggings_count_cache
+
+    # If we've reached here, then the value has changed, and we need to make
+    # sure that the new value is written to the database.
     REDIS_GENERAL.sadd("tag_update", id)
```

Comment:
It's still a set either way, so it won't be called twice (at least until after it's been processed). But suppose that the taggings count hasn't changed, and someone views the Edit page for a small tag (or even a large one). The taggings count will be recomputed and it will be added to REDIS and the `tag_update` set for an update, even though absolutely nothing has changed since the last time the taggings count was written to the database.